app/models/subject.rb

```diff
@@ -1,7 +1,8 @@
 class Subject < ApplicationRecord
+  acts_as_paranoid
   belongs_to :user, class_name: User.name
-  has_many :tests, dependent: :destroy
-  has_many :questions, dependent: :destroy
+  has_many :tests, dependent: nil
```

Comment:
```suggestion
  has_many :tests, dependent: :nullify
```
ri chứ nhỉ

Reply:
em muốn sau vẫn gọi được test.subject nên để là nil cho nó không null cái khóa ngoại trên bảng test ạ.
với trong model cũng để subject_id là not null nữa ạ :v